### PR TITLE
Pt 166698281 allow sc ws to disconnect 2

### PR DESCRIPTION
**Pull request by uwiger**
Same as #2600 , but rebased on top of #2596 (just merged).
Deps updated to account for the related `aeserialization` PR being merged, changing the commit hash.

### CODE REVIEW

**Comment by lucafavatella** [comment]
@uwiger I read "Conflicting files apps/aechannel/src/aesc_fsm.erl". Does this need new rebase?

**Comment by uwiger** [comment]
@lucafavatella yes, git was unhappy with the state of some of the deleted .puml files. I did a new rebase.

File: rebar.lock [hunk 1]
```diff
@@ -18,7 +18,7 @@
   0},
  {<<"aeserialization">>,
   {git,"https://github.com/aeternity/aeserialization.git",
-      {ref,"3416ff5c1b4f6401bc211106063717ea1bbb1bd2"}},
+      {ref,"4a072977745571ceb6cf34e6387a96b808ae20b3"}},
```
Comment (lucafavatella) [inline rebar.lock]: https://github.com/aeternity/aeserialization/compare/3416ff5c1b4f6401bc211106063717ea1bbb1bd2...4a072977745571ceb6cf34e6387a96b808ae20b3
> tag(channel_client_reconnect_tx) -> 575;
> ...

Comment (lucafavatella) [reply rebar.lock]: Shall this require doc update in https://github.com/aeternity/protocol/blob/master/serializations.md ? 🤔

Comment (lucafavatella) [reply rebar.lock]: Is this tx anyhow under consensus?

File: apps/aechannel/src/aesc_client_reconnect_tx.erl [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+state_hash(#channel_client_reconnect_tx{}) ->
+    error(invalid).
+
+-spec version(tx()) -> non_neg_integer().
+version(_) ->
+    ?INITIAL_VSN.
+
+-spec valid_at_protocol(aec_hard_forks:protocol_vsn(), tx()) -> boolean().
+valid_at_protocol(_Protocol, _Tx) ->
+    true.
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_client_reconnect_tx.erl]: 🤔

Comment (lucafavatella) [reply apps/aechannel/src/aesc_client_reconnect_tx.erl]: I see `channel_client_reconnect_tx` has gas_limit 0 so not included in microblock candidate though I do not understand how this is prevented from entering mempool.

File: apps/aechannel/src/aesc_client_reconnect_tx.erl [hunk 3]
```diff
@@ -0,0 +1,193 @@
+%%% @copyright 2019, Aeternity Anstalt
+%%% @doc
+%%%    Module defining an offchain tx for reconnecting a client to the SC FSM.
+%%%    The tx contains the information needed to locate the right FSM, including
+%%%    the client's public key. It must be signed by the client, proving that it
+%%%    possesses the private key (or other signing capability) for the account.
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_client_reconnect_tx.erl]:
```suggestion
%%%    possesses the private key (or other signing capability) for the account.
%%%
%%%    This is a transaction so to ease signing for the clients of the SC FSM.
```

File: apps/aechannel/src/aesc_client_reconnect_tx.erl (hunk 2 above)
Comment (lucafavatella) [reply apps/aechannel/src/aesc_client_reconnect_tx.erl]:
```suggestion
    false.
```

File: rebar.lock (hunk 1 above)
Comment (uwiger) [reply rebar.lock]: > Is this tx anyhow under consensus?

No, like the `aesc_offchain_tx`, it is not designed to go on-chain: the `fee` is hard-coded to zero, as is the `ttl`, and `aesc_offchain_tx:process()` will raise an exception.

Comment (uwiger) [reply rebar.lock]: > Shall this require doc update in https://github.com/aeternity/protocol/blob/master/serializations.md ? thinking

Working on documentation now ([ticket](https://www.pivotaltracker.com/story/show/167617278))

**Comment by uwiger** [comment]
> The handling of the relation of lifecycle of FSM and Noise connection to peer may need some cleanup for readability though out of scope for this PR.

https://www.pivotaltracker.com/story/show/167617973

File: apps/aechannel/src/aesc_fsm.erl [hunk 4]
```diff
@@ -15,8 +15,9 @@
 %% API
 -export([ start_link/1
         , attach_responder/2      %% (fsm(), map())
-        , client_died/1           %% (fsm())
         , close_solo/1
+        , reconnect_client/2        %% (fsm(), signed_tx())
```
Comment (grzegorz225) [inline apps/aechannel/src/aesc_fsm.erl]: Remove 2 spaces before comment in order to align the comment with the rest.

File: apps/aechannel/src/aesc_fsm.erl [hunk 5]
```diff
@@ -15,8 +15,9 @@
...
+        , reconnect_client/3        %% (fsm(), Client :: pid(), signed_tx())
```
Comment (grzegorz225) [inline apps/aechannel/src/aesc_fsm.erl]: Same as above

File: apps/aehttp/test/aehttp_sc_SUITE.erl [hunk 6]
```diff
@@ -2402,6 +2419,91 @@ sc_ws_basic_open_close_server(Config0) ->
...
+            end
+        end,
+    %% sender initiates an update
+    ws_send_tagged(ConnPid, <<"channels.update.new">>,
+                   #{from => aeser_api_encoder:encode(account_pubkey, FromPubkey),
+                     to => aeser_api_encoder:encode(account_pubkey, ToPubkey),
+                     amount => 1}, Config),
+
+    %% starter signs the new state
+
```
Comment (grzegorz225) [inline apps/aehttp/test/aehttp_sc_SUITE.erl]: Is this vertical space necessary?

**Pull request merged**